client/src/containers/CampaignAdmin/CampaignAdmin.js

```diff
@@ -119,7 +120,8 @@ class CampaignAdmin extends SilverStripeComponent {
           // open the new campaign in edit mode after save completes
           const sectionUrl = this.props.sectionConfig.url;
           const id = response.record.id;
-          this.props.router.push(`${sectionUrl}/set/${id}/edit`);
+          this.props.campaignActions.setNewCampaignCreated(true);
```

Comment:
Since the `id` is available, it would be more extensible to include the record or even just the id in the record, rather than just a boolean.
In this case, you could ignore the `is` or `has` prefix I mentioned below. But try to keep the naming consistent still `setNewItem` is perfectly fine.